Makefile

```diff
@@ -1,157 +1,282 @@
+	sed -e '1 s/\(^FROM\)/FROM --platform=\$$\{BUILDPLATFORM\}/; t' -e ' 1,// s//FROM --platform=\$$\{BUILDPLATFORM\}/' Dockerfile > Dockerfile.cross
+	- docker buildx create --name project-v3-builder
+	docker buildx use project-v3-builder
+	- docker buildx build --push --platform=$(PLATFORMS) --tag ${IMG} -f Dockerfile.cross .
+	- docker buildx rm project-v3-builder
+	rm Dockerfile.cross
+
+.PHONY: helm-chart
+helm-chart: helm ## Build helm chart
+	@mkdir -p build
+	$(HELM) package -d build/ deploy/charts/vault-operator
+
+.PHONY: build-artifacts
+build-artifacts: docker-build helm-chart
+build-artifacts: ## Build docker image and helm chart
+
+##@ Deployment
+
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
+
+.PHONY: up
+up: kind ## Start kind development environment
+	$(KIND) create cluster --name vault-operator
+
+.PHONY: down
+down: kind ## Destroy kind development environment
+	$(KIND) delete cluster --name vault-operator
+
+.PHONY: import-image
+import-image: kind ## Import operator docker image to kind image repository
+	$(KIND) load docker-image ${IMG} --name vault-operator
+
+.PHONY: import-images-acceptance
+import-images-acceptance: import-image ## Import docker images required for acceptance tests to kind image repository
+	docker pull ghcr.io/banzaicloud/bank-vaults:$(TEST_BANK_VAULTS_VERSION)
+	docker pull vault:$(TEST_VAULT_VERSION)
+	docker tag vault:$(TEST_VAULT_VERSION) vault:latest
+
+	$(KIND) load docker-image ghcr.io/banzaicloud/bank-vaults:$(TEST_BANK_VAULTS_VERSION) --name vault-operator
+	$(KIND) load docker-image vault:latest --name vault-operator
+
+.PHONY: clean
+clean: undeploy ## Clean operator resources from a Kubernetes cluster
+
+.PHONY: install
+install: generate-manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build deploy/crd | kubectl apply -f -
+
+.PHONY: uninstall
+uninstall: generate-manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+	$(KUSTOMIZE) build deploy/crd | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
+
+.PHONY: deploy
+deploy: generate-manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	cd deploy/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	$(KUSTOMIZE) build deploy/default | kubectl apply -f -
+
+.PHONY: undeploy
+undeploy: kustomize ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+	$(KUSTOMIZE) build deploy/default | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
+
+##@ Dependencies
+
+## Location to install dependencies to
+LOCALBIN ?= $(shell pwd)/bin
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+KUSTOMIZE ?= $(LOCALBIN)/kustomize
```

Comment:
This is still possible, but now it is per-tool basis, e.g. `export LICENSEI=path/to/licensei`

Reply:
This enforces the versioning for reproducibility, e.g. the autogeneration did not work at all because it kept mixing data from nix and my OS.

Reply:
I prefer the default to fallback to what's in the PATH and only use locally installed binaries if they are present. That's exactly what the PATH is for.

Reply:
Resolved. The Makefile cannot change/use PATH variable, so it has to dynamically deduce this from the shell. The reason behind this is that I wanted to avoid developers having to set PATH variable every time they want to work with BV makefile, and same goes to the automated tools. This was a compromise between both.